modules/saml/lib/IdP/SQLNameID.php

```diff
@@ -13,40 +15,103 @@
  */
 class SQLNameID
 {
+    const TABLE_VERSION = 1;
+    const DEFAULT_TABLE_PREFIX = '';
+    const TABLE_SUFFIX = '_saml_PersistentNameID';
+
+
     /**
-     * Create NameID table in SQL, if it is missing.
-     *
-     * @param \SimpleSAML\Store\SQL $store  The datastore.
-     * @return void
+     * @param string $query
+     * @param array $params
+     * @param array $config
+     * @return \PDOStatement
      */
-    private static function createTable(Store\SQL $store)
+    private static function query($query, array $params = [], array $config = [])
     {
-        if ($store->getTableVersion('saml_PersistentNameID') === 1) {
-            return;
+        if (!empty($config)) {
+            $database = Database::getInstance(Configuration::loadFromArray($config));
+            if (stripos($query, 'SELECT') === 0) {
+                $stmt = $database->read($query, $params);
```

Comment:
I'm a bit worried about the newly added methods... As you can see, Psalm-tests fail, because `Database::read` will not return a `PDOStatement` as you are assuming here and everywhere else in this file..

Reply:
You're right, `Database::write` returns `int|bool`, but here on write operations the results are being ignored anyway, so a return type of `PDOStatement|int|bool` would fix this; anyway, I've split the `query` method into `read` and `write`, basically mimicking `Database` using `Storage`, and got rid of the `if (stripos($query, 'SELECT') === 0)`.